rules_java_gapic/java_gapic_pkg.bzl

```diff
@@ -86,18 +86,34 @@ def _gapic_pkg_tar_impl(ctx):
         mkdir -p {package_dir_path}/{tar_cd_suffix}/{tar_prefix}/samples/snippets/generated/
         unzip -q ./$s -d {package_dir_path}/{tar_cd_suffix}/{tar_prefix}/samples/snippets/generated/
     done
-
     mkdir -p {package_dir_path}
     for dep in {deps}; do
         tar -xzpf $dep -C {package_dir_path}
     done
-    cd {package_dir_path}/{tar_cd_suffix}
 
+    #TODO: Use awk to extract classes inside java class. Such as: awk '/public class/' Classes.java
+    if [[ {package_dir_path} == *proto-google-cloud-* ]] || [[ {package_dir_path} == *gapic-google-cloud-* ]] || [[ {package_dir_path} == *grpc-google-cloud-* ]]; then
```

Comment:
Let's update this PR to:

- [ ] Also apply to Showcase client generation
- [ ] Invoke the showcase native image tests (free if rebasing on #1833)
- [ ] Remove the RunReachabilityHandlersConcurrently argument
- [ ] Remove the Feature classes using reachability handlers